packages/@uppy/status-bar/src/StatusBar.tsx

```diff
@@ -216,27 +232,30 @@ export default class StatusBar extends UIPlugin {
       hidePauseResumeButton: this.opts.hidePauseResumeButton,
       hideCancelButton: this.opts.hideCancelButton,
       hideAfterFinish: this.opts.hideAfterFinish,
+      // ts-expect-error TODO: remove this in 4.x branch
       isTargetDOMEl: this.isTargetDOMEl,
     })
   }
 
-  onMount () {
+  onMount(): void {
     // Set the text direction if the page has not defined one.
     const element = this.el
-    const direction = getTextDirection(element)
+    const direction = getTextDirection(element!)
     if (!direction) {
-      element.dir = 'ltr'
+      element!.dir = 'ltr'
     }
   }
 
-  #onUploadStart = () => {
+  #onUploadStart = (): void => {
     const { recoveredState } = this.uppy.getState()
 
     this.#previousSpeed = null
     this.#previousETA = null
     if (recoveredState) {
-      this.#previousUploadedBytes = Object.values(recoveredState.files)
-        .reduce((pv, { progress }) => pv + progress.bytesUploaded, 0)
+      this.#previousUploadedBytes = Object.values(recoveredState.files).reduce(
+        (pv, { progress }) => pv + (progress.bytesUploaded as number),
```

Comment:
```suggestion
        (pv, { progress }) => pv + (progress.bytesUploaded as number), // todo type progress.bytesUploaded propertly
```

Reply:
I would argue that `bytesUploaded` _is_ typed properly, and TS is not wrong in throwing an error regarding `false` not being a number, but in this case it's perfectly to let JS coerce `false` to `0`.

Reply:
I'm willing to take that risk, I see no reason we would start using strings for `bytesUploaded`, and if we did it would be fairly obvious that the status-bar was broken